edge-hub/core/test/Microsoft.Azure.Devices.Edge.Hub.E2E.Test/Cloud2DeviceTest.cs

```diff
@@ -128,8 +85,16 @@ public async void Receive_C2D_SingleMessage_AfterOfflineMessage_ShouldSucceed()
                 // Dummy ReceiveAsync to ensure mqtt subscription registration before SendAsync() is called on service client.
                 await deviceClient.ReceiveAsync(TimeSpan.FromSeconds(1));
                 await deviceClient.CloseAsync();
-                // wait for the connection to be closed on the Edge side
-                await Task.Delay(TimeSpan.FromSeconds(30));
+
+                // Wait for the connection to be closed on the Edge side by checking device connection state
+                Task timerTask = Task.Delay(TimeSpan.FromSeconds(60));
```

Comment:
Why bump to 60 seconds?

Reply:
Guess this is now bumped to 4 minutes? Will this be enough time to ensure the test is not flaky?

Reply:
I thought 60s would help, but it was still failing so just trying to see if 4min would help, but it was still failing. I found the issue and I think with the current change is stable, still running a few more times. So it needs to wait for connect before it disconnects. I opened a bug for the root cause issue, it is a corner case if device connects and quickly disconnects before edgeHub has time to connect to cloud, then cloud connection is not closed and c2d messages are sent and edgehub can't send it to device since it was disconnected.